src/components/split_button/split_button_control.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, {
+  FunctionComponent,
+  ButtonHTMLAttributes,
+  ReactNode,
+} from 'react';
+
+import { CommonProps } from '../common';
+
+import {
+  ButtonColor,
+  ButtonSize,
+  OuiButton,
+  OuiButtonIcon,
+  OuiButtonIconColor,
+  OuiButtonProps,
+} from '../button';
+
+// this intersection still does not satisfy OuiButtonIconColor
+// https://github.com/opensearch-project/oui/issues/1196
+export type OuiSplitButtonColor = OuiButtonIconColor & ButtonColor;
+
+export interface OuiSplitButtonControlProps
+  extends CommonProps,
+    Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'color'> {
+  fullWidth?: boolean;
+  isLoading?: boolean;
+
+  fill?: boolean;
+
+  /**
+   * Color of buttons and options
+   */
+  color?: OuiSplitButtonColor;
+
+  /**
+   * Use size `s` in confined spaces
+   */
+  size?: ButtonSize;
+
+  /**
+   * Click handler of Primary button
+   */
+  onClick?: () => void;
+
+  /**
+   * Click handler for drop-down button -- used by SplitButton to control
+   * OuiPopover
+   */
+  onDropdownClick?: () => void;
+
+  /**
+   * Handle key-events for dropdown control
+   */
+  onKeyDown?: (event: React.KeyboardEvent<HTMLButtonElement>) => void;
+
+  /**
+   * Optional additional props to send to Primary Button
+   */
+  propsForPrimaryButton?: OuiButtonProps;
+
+  /**
+   * Optional additional props to send to Dropdown Button
+   */
+  propsForDropdownButton?: OuiButtonProps;
+
+  /**
+   * Content of Primary (left-side) button
+   */
+  children: ReactNode;
+}
+
+export const OuiSplitButtonControl = ({
+  fill,
+  size,
+  color = 'primary',
+  children,
+  fullWidth,
+  onClick,
+  onDropdownClick,
+  onKeyDown: onSelectKeydown,
+  propsForDropdownButton,
+  propsForPrimaryButton,
+}: OuiSplitButtonControlProps): ReturnType<
+  FunctionComponent<OuiSplitButtonControlProps>
+> => {
+  const iconDisplay = fill ? 'fill' : 'base';
+
+  return (
+    <div className="ouiSplitButtonControl">
+      <OuiButton
+        className="ouiSplitButtonControl--primary"
+        fill={fill}
+        color={color}
+        size={size}
+        fullWidth={fullWidth}
+        onClick={onClick}
+        onKeyDown={onSelectKeydown}
+        {...propsForPrimaryButton}>
+        {children}
+      </OuiButton>
+      <OuiButtonIcon
+        display={iconDisplay}
+        //@ts-ignore - typedef conflict between ButtonColor, OuiButtonIconColor
+        // https://github.com/opensearch-project/oui/issues/1196
```

Comment:
Per conversation w/ @ashwin-pc , we decided the prudent option was to except this and track an issue for the typedef conflict.

Reply:
it happens because of https://github.com/opensearch-project/oui/blob/59edc961c7005491fec6de6f3cb45cf755da2ab9/src/components/split_button/split_button_control.tsx#L123, i commented on your issue

Reply:
@pjfitzgibbons It seems like the click action triggers two independent button animations - 1 on the clicked individual inner button and 1 on the container. Currently it appears the way this split button has been built is using two separate buttons (a Button with a label and a Button with a down arrow icon wrapped in a container that inherits button animation properties. I would opt to remove the animation all together to avoid this unwanted animation behavior - this animation works best when it's a singular button and maybe not so well as a split. @kgcreative any thoughts?

Reply:
Yeah,to add to this, it looks like each individual element has all the chrome of the parent (individual borders, corner radii, etc). That is firmly not in the spirit of this component.The outer border and the animation when you click the button should apply to the outer container. When you click the button, you should get the depress behavior on the whole element. When you click the drop down, there should be no animation. This is definitely a 2.13 follow-up. If I have some time I may do a couple of quick animations to illustrate the difference, but when you look at Carlos' design, the border between the buttons is a single vertical bar, not two left and right borders that come together.

Reply:
A note for 2.13 follow up : re-using OuiButton/OuiButtonIcon may not be final solution then, if we want to draw the chrome of the component around the containing element of both buttons.

Reply:
I think I have a solution in local dev.

If one of you could tell me exactly how to achieve the color of the hairline for both light and dark modes?
I've been trying a bunch of things around shading and I think I don't know enough how SASS works.
That would be the last detail that I need to get this working 100%.

Your suggestions would be incredibly appreciated.

@kgcreative @canascar @joshuali925 @ashwin-pc

Reply:
@joshuali925 Nice catch! @pjfitzgibbons can we fix this then. Just moving the color prop to after `propsForDropdownButton` should fix this.